### PR TITLE
refactor(kibana): unexport KibanaScopedClient version accessors

### PR DESCRIPTION
**Pull request by tobio**
## Summary

Removes `KibanaScopedClient.ServerVersion()` and `KibanaScopedClient.ServerFlavor()` from the public surface and migrates every caller to the serverless-safe version-gating primitives that already exist (`EnforceMinVersion`, `EnforceVersionCheck`, `entitycore.WithVersionRequirements`). Closes the type-level hazard that the recently archived `fix-serverless-version-gating` change had to remediate post-hoc — a future Kibana resource can no longer accidentally break serverless because the unsafe API will not compile.

OpenSpec change: `kibana-scoped-client-unexport-version-accessors`.

### Migration patterns applied
- **Pattern A — `WithVersionRequirements` on the model** (8 callers): `security_enable_rule {read,update,delete}`, `prebuilt_rules {read,update}`, `slo {create,update}`, `connectors/create`. Pure gates moved onto the entity model; the Kibana resource envelope (or explicit `entitycore.EnforceVersionRequirements` for resources still on the SDK base) enforces them automatically.
- **Pattern B — `EnforceMinVersion` boolean for feature toggles** (2 callers): `synthetics/parameter/delete` and `agentbuilderagent/data_source`. Serverless naturally selects the newer API path.
- **Pattern C — capability struct** (alertingrule): new `alertingRuleFeatures` + `resolveAlertingRuleFeatures`, mirroring the `agentpolicy.features` precedent. `toAPIModel` no longer takes a `*version.Version`.
- **Build fallout** (caught while deleting the public methods): `internal/fleet/integration/create.go` and `internal/acctest/acctest.go` also called the methods and were migrated to `EnforceMinVersion`.

### Tests
- New `GetVersionRequirements` unit tests for `security_enable_rule`, `prebuilt_rules`, `slo`, `connectors` (including all four condition combinations for SLO and the `prevent_initial_backfill = false` edge case).
- New `TestResolveAlertingRuleFeatures` covering server-below-all, server-above-all, and serverless-short-circuit.
- `internal/clients/kibana_scoped_client_test.go` rewritten so the missing-endpoint / below-min / at-min / above-min / serverless-short-circuit / malformed-version / status-API-error scenarios are exercised through the public `EnforceMinVersion` and `EnforceVersionCheck` methods.
- `internal/clients/provider_client_factory_test.go` factory tests renamed to `TestKibanaScopedClient_EnforceMinVersion_ViaFactory_Stateful` and `..._ServerlessShortCircuit`.
- Acceptance test sweep (`rg "ServerVersion |ServerFlavor" internal/kibana/**/acc_test.go`) confirmed no Kibana acc-test references remain.

### Specs
- `openspec/specs/provider-client-factory/spec.md`: Kibana scoped client contract now names `EnforceMinVersion`, `EnforceVersionCheck`, and `entitycore.WithVersionRequirements` as the only supported version-gating surfaces, and explicitly forbids public `ServerVersion()`/`ServerFlavor()`.
- `openspec/specs/elasticsearch-client-pf-diagnostics/spec.md`: `KibanaScopedClient` method enumeration updated to `EnforceMinVersion` + `EnforceVersionCheck`; the `ServerVersion` pass-through scenario is replaced with explicit forbid-raw-accessors and `EnforceVersionCheck` PF-diag scenarios.

## Changelog
Customer impact: none

## Test plan

- [x] `make check-openspec` — pass
- [x] `make build` — pass
- [x] `go test ./internal/kibana/... ./internal/clients/... ./internal/entitycore/...` — pass (pre-existing env-dependent failures in `security_enable_rule/acc_test.go` only; not regressions)
- [x] Targeted acceptance tests against local stack — pass for `security_enable_rule`, `prebuilt_rules`, `slo`, `connectors`, `synthetics/parameter`, `agentbuilderagent`; 2 flaky `alertingrule` failures observed on `last_execution_date`/`last_execution_status` drift, unrelated to this refactor
- [ ] Awaiting GitHub Actions CI on the PR branch

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: internal/kibana/alertingrule/features.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+	supportsFlapping, diags := client.EnforceMinVersion(ctx, flappingMinSupportedVersion)
+	if diags.HasError() {
+		return alertingRuleFeatures{}, diags
+	}
+
+	supportsFlappingEnabled, diags := client.EnforceMinVersion(ctx, flappingEnabledMinSupportedVersion)
+	if diags.HasError() {
+		return alertingRuleFeatures{}, diags
+	}
```
[the remark this answers is not in the document]
Comment (tobio) [reply internal/kibana/alertingrule/features.go]: Addressed in 43096fc7f0033800817355517aa2311b3b6f11b3: resolve all alerting rule feature flags from a single EnforceVersionCheck call (one /api/status fetch).

File: internal/kibana/alertingrule/models.go [hunk 2]
```diff
@@ -330,90 +330,88 @@ var (
...
+	// alert_delay is only supported from v8.13+
+	if typeutils.IsKnown(m.AlertDelay) && !m.AlertDelay.IsNull() {
+		if !features.SupportsAlertDelay {
+			diags.AddError(
+				"alert_delay is only supported for Elasticsearch v8.13 or higher",
+				"alert_delay is only supported for Elasticsearch v8.13 or higher",
+			)
+			return models.AlertingRule{}, diags
+		}
+	}
```
[the remark this answers is not in the document]
Comment (tobio) [reply internal/kibana/alertingrule/models.go:358]: Addressed in 43096fc7f0033800817355517aa2311b3b6f11b3: alert_delay error message now references Kibana v8.13 to match sibling messages.
